### PR TITLE
MRG : source reconstruction on EEG using fsaverage brain

### PR DESCRIPTION
**Pull request by agramfort**
this PR aims to cover a long standing issue with MNE which is to do EEG source reconstruction using a template brain here fsaverage.

Limitation of the current PR:
- very coarse narrative doc
- has no cross-links with other tutorials
- only tests with one montage file 'standard_1005'
- data contains induced power so current evoked pipeline is not appropriate.
- needs `mne.datasets.fetch_fsaverage_3layer_bem` or so

@drammock any thoughts on how to integrate this in the new in the making documentation?

@massich this is a good test for the new digitization refactoring

### CODE REVIEW

File: tutorials/plot_eeg_no_mri.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+src = mne.setup_source_space(subject, spacing='oct6',
+                             subjects_dir=subjects_dir, add_dist=False)
+print(src)
+
+mne.viz.plot_bem(subject=subject, subjects_dir=subjects_dir,
+                 brain_surfaces='white', src=src, orientation='coronal')
+
+conductivity = (0.3, 0.006, 0.3)  # for three layers
+model = mne.make_bem_model(subject=subject, ico=4,
```
Comment (mmagnuski) [inline tutorials/plot_eeg_no_mri.py]: the difference between `make_bem_model` and `make_bem_solution` may be confusing to a newcomer. Maybe add a short comment or just rename the `model` variable to `bem_surfaces`?

**Comment by agramfort** [comment]
we should put here a link to the forward tutorial rather than duplicating
the narrative. I won't touch at this before the sprint next week

>

File: tutorials/plot_eeg_no_mri.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                              method=method, pick_ori=None,
+                              return_residual=True, verbose=True)
+
+###############################################################################
+# Now plot the source estimates
+# -----------------------------
+
+vertno_max, time_max = stc.get_peak(hemi='rh')
+
+subjects_dir = data_path + '/subjects'
```
Comment (mmagnuski) [inline tutorials/plot_eeg_no_mri.py]: this variable is already present at the top

File: tutorials/plot_eeg_no_mri.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    epochs, tmax=0., method=['shrunk', 'empirical'], rank='full')
+
+# make an MEG inverse operator
+info = evoked.info
+inverse_operator = make_inverse_operator(info, fwd, noise_cov,
+                                         loose=0.2, depth=0.8)
+
+
+method = "dSPM"
+snr = 3.
```
Comment (mmagnuski) [inline tutorials/plot_eeg_no_mri.py]: maybe a short explanation why we set snr to 3?

File: tutorials/plot_eeg_no_mri.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+ch_names = [c.replace('.', '') for c in raw.ch_names]
+raw.rename_channels({old: new for old, new in zip(raw.ch_names, ch_names)})
+
+# Read and set the EEG electrode locations
+montage = mne.channels.read_montage('standard_1005', ch_names=raw.ch_names,
+                                    transform=True)
+raw.set_montage(montage)
+
+
+subject = 'fsaverage'
```
Comment (mmagnuski) [inline tutorials/plot_eeg_no_mri.py]: I'd put subject at the top along with `subjects_dir`

File: tutorials/plot_eeg_no_mri.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+# Now plot the source estimates
+# -----------------------------
+
+vertno_max, time_max = stc.get_peak(hemi='rh')
+
+subjects_dir = data_path + '/subjects'
+surfer_kwargs = dict(
+    hemi='rh', subjects_dir=subjects_dir, views='lateral',
+    initial_time=time_max, time_unit='s', size=(800, 800), smoothing_steps=5)
+brain = stc.plot(**surfer_kwargs)
```
Comment (mmagnuski) [inline tutorials/plot_eeg_no_mri.py]: why separate into `surfer_kwargs dict`? Would be simpler to have these directly in the function call.

**Comment by agramfort** [comment]
it's WIP so we don't loose this. If you have time to improve please
push directly to this branch.

**Comment by mmagnuski** [comment]
Circle errors on the example:
```python-traceback
Traceback (most recent call last):
  File "/home/circleci/.local/lib/python3.7/site-packages/sphinx_gallery/gen_rst.py", line 398, in _memory_usage
    multiprocess=True)
  File "/home/circleci/.local/lib/python3.7/site-packages/memory_profiler.py", line 336, in memory_usage
    returned = f(*args, **kw)
  File "/home/circleci/.local/lib/python3.7/site-packages/sphinx_gallery/gen_rst.py", line 389, in __call__
    exec(self.code, self.globals)
  File "/home/circleci/project/tutorials/plot_eeg_no_mri.py", line 58, in <module>
    subjects_dir=subjects_dir, trans=trans_fname)
  File "</home/circleci/project/mne/externals/decorator.py:decorator-gen-117>", line 2, in plot_alignment
  File "/home/circleci/project/mne/utils/_logging.py", line 89, in wrapper
    return function(*args, **kwargs)
  File "/home/circleci/project/mne/viz/_3d.py", line 815, in plot_alignment
    % (subject, '\n'.join(try_fnames)))
OSError: No head surface found for subject fsaverage after trying:
/home/circleci/mne_data/MNE-sample-data/subjects/fsaverage/bem/outer_skin.surf
/home/circleci/mne_data/MNE-sample-data/subjects/fsaverage/bem/flash/outer_skin.surf
/home/circleci/mne_data/MNE-sample-data/subjects/fsaverage/bem/fsaverage-head.fif
```

**Comment by mmagnuski** [comment]
@agramfort Sure, I can try, but I had a lot of problems before trying to push to other people's PRs. If you also have a tutorial on that, that would be helpful. :)

**Comment by mmagnuski** [comment]
Regarding the error: is it possible that mne's fsaverage ships without head surface? I remember I had to construct it when using fsaverage for source localization.

**Comment by agramfort** [comment]
yes a new fsaverage dataset is necessary or at least an update of sample
dataset to add these missing surfaces. I had to compute them locally.

>

**Comment by larsoner** [comment]
If you upload the surfaces I can update the dataset. Or if it's as simple as `mne watershed_bem` I'll run it locally and update.

**Comment by agramfort** [comment]
yes it's just running mne watershed_bem
we can add this to sample but I think it could be smart to have a tiny
dataset with just fsaverage
for those we want to do EEG with MRI templates

File: tutorials/plot_eeg_no_mri.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+"""
+EEG source reconstruction with a template MRI
+=============================================
+
+This tutorial explains how to perform source reconstruction using
+EEG using template MRI subject. We use here the fsaverage brain
+provided by freesurder.
```
Comment (wmvanvliet) [inline tutorials/plot_eeg_no_mri.py]: *freesurfer

**Comment by agramfort** [comment]
+1 for a specific fetcher like mne.datasets.fetch_fsaverage_3layer_bem
but I would not assume that sample is present to have it working

>

**Comment by massich** [comment]
This tuto does something similar to what we have in #5836 but with `fsavarege` rather than `sample`. I can take over and the missing files so that this PR renders in circle.

**Comment by massich** [comment]
here is the rendering https://12255-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/plot_eeg_no_mri.html?highlight=plot_eeg_no_mri

**Comment by massich** [comment]
The current solution requires to download 1.8Gb we could either allow for:
1) Partial download of the sample
2) split sample and fsaverage and,
   2a) ship them separately
   2b) ship them twice

**Comment by drammock** [comment]
@agramfort for now, putting it in the root of `tutorials` (as you've done) is fine.  This PR will almost surely get merged before the big tutorials overhaul, and I'll rebase against it and move this new tutorial as necessary at that time.

File: examples/visualization/plot_montage.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+data_path = mne.datasets.sample.data_path()
+subjects_dir = op.join(data_path, 'subjects')
+trans_fname = op.join(op.dirname(mne.__file__), "data", "fsaverage",
+                      "fsaverage-trans.fif")
+
+###############################################################################
+# check all montages
+#
+
+fix_units = {'EGI_256':'cm', 'GSN-HydroCel-128':'cm', 'GSN-HydroCel-129':'cm',
```
Comment (mmagnuski) [inline examples/visualization/plot_montage.py]: EGI_256 is not in cm, take a look at [this comment](https://github.com/mne-tools/mne-python/pull/5472#issuecomment-483059750): the distance for each channel from the center is exactly 1. for EGI_256.

Comment (massich) [reply examples/visualization/plot_montage.py]: I missed your comment. The way you figured out the distance is smarter than the way I did it. 

The code was pulled from an early stage of #5836 and yes `EGI_256` it's broken.  Now that we are at it, when load `EGI_256` with `eeg=['original', 'projected']` you can see that the original do have a 'brainish' shape but the projected does not.

Either way, we decided with @agramfort to add the example ploting all helmets i a different PR. (I'll ping you there)

**Comment by larsoner** [comment]
I think the way to go on this units business is:

1. Ensure all our montages are spherical with origin `(0., 0., 0.)` and properly defined LPA/RPA/Nasion (I think they are; if they are not, it's probably a bug)
2. On loading a montage, resize the sphere defining the electrodes/LPA/RPA/Nasion to be 0.08 m radius (matches the default we use for `make_sphere_model`)
3. Eventually expose a parameter to allow users to set this radius (they probably have a rough estimate for their participant's head size, e.g. adult/child)

Then we don't have to care what units we add our montages in, which makes maintainability easier because we can keep the original versions.

**Comment by agramfort** [comment]
this looks wrong https://12259-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/plot_eeg_no_mri.html

can you threshold lower?

@massich compare to how it looks with sample EEG data.

**Comment by mmagnuski** [comment]
@agramfort @massich Are you sure it is better to finish the tutorial with sensitivity plot not actual source reconstruction? I know we could point to longer tutorials at the end of this one but I think it would be more complete if some source reconstruction was shown, not only the sensitivity.

@larsoner :+1:, some EGI montages are not precisely spherical because they seem to follow a little more realistic head shape (this is mostly for the channels residing on the cheeks, but I think at least one is also a bit more ellipsoid)

**Comment by mmagnuski** [comment]
@larsoner BTW, do we plan to store the info on units and radius in the current montage text files or is the plan to always guess? I know guessing would be inevitable anyway as users may provide various montages, even some without fiducials, but still it would be nice to allow reading this from the montage text file.

**Comment by larsoner** [comment]
I think that if the file format allows reading units we should do it. My argument is that, in the end, ideally it wouldn't matter whether or not they were provided, i.e. we would treat all montages the same way / scale them to the same desired size (and shape!) when requested.

**Comment by massich** [comment]
@mmagnuski to me the tutorial could finish once you have the forward the rest is just to have some nice plot to see which kind of info you can get. @agramfort proposed to move to the sensitivity map to keep it shorter (I'll let @agramfort elaborate)

**Comment by agramfort** [comment]
yes the idea is that the bci dataset we have is not ideal for ERP so
standard pipeline. And once you have forward you go to go with any example
we have elsewhere. sensitivity maps are great to end to demo that it works

**Comment by mmagnuski** [comment]
Ah, right, if the dataset is not good for ERP I get it. Then it would be good to just mention that now one can progress along the route sketched in other tutorials to source localize.

File: tutorials/plot_eeg_no_mri.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+"""
+EEG source reconstruction with a template MRI
+=============================================
+
+This tutorial explains how to perform source reconstruction using
```
Comment (mmagnuski) [inline tutorials/plot_eeg_no_mri.py]: I'd also change the title and description a bit to highlight that the tutorial shows how to prepare all the necessary components for source reconstruction but does not actually perform reconstruction.

**Comment by agramfort** [comment]
feel free to push directly here. We wanted to have the ball rolling. FYI in
our team this use case does not exist as we always have MRIs even if we
only have EEG.

>

**Comment by mmagnuski** [comment]
looks good: https://12268-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/plot_eeg_no_mri.html

**Comment by mmagnuski** [comment]
@agramfort I'd be happy to push here but I had troubles with that some time ago (@massich was trying to help me with setting up pushing to PRs) and since that time I never had the courage to go though trying to set it up again (for some reason I couldn't get it working although I tried a few ways). I may have some patiance to try with that tomorrow. :)
Good for you if you always have MRIs, :) I know many people that don't and a tutorial like that would be helpful.

**Comment by mmagnuski** [comment]
:+1:, me and pushing to other PRs seem not to be destined to meet :)

**Comment by massich** [comment]
@mmagnuski you can use the new version of [hub](https://hub.github.com).
Then `hub pr checkout 6146` and you should be able to push.

**Comment by mmagnuski** [comment]
@massich You were advicing me on hub the last time we tried and I had problems with it too. But I came back to hub lately and I'll try it next time. For now I don't have anything to suggest here, the PR is good as it is, so no need for me to try pushing here just for the sake of it. :)

**Comment by agramfort** [comment]
any improvement of narrative and documentation is good to take :)

**Comment by massich** [comment]
Does anyone know why the last commit breaks? it takes the wrong file and the hash no longer matches. What I don't understand is why the previous commit was not failing.

**Comment by wmvanvliet** [comment]
@larsoner put the old file back. The change was breaking the hash for all users of the current mne-python release.

**Comment by massich** [comment]
Then we should revert 32e1452

**Comment by larsoner** [comment]
Yes, I have a PR coming. The state before I re-uploaded the old `sample` dataset was that anyone not on `master` as of this morning could no longer use our `mne.datasets.sample.data_path()` to download sample data, which is quite bad.

Now it's in the state that anyone on `master` as of today can't, and everyone else can (which is bad, but preferable to the other situation). But the upcoming PR should fix the `master` users.

File: mne/datasets/utils.py [hunk 9]
```diff
@@ -252,7 +252,7 @@ def _data_path(path=None, force_update=False, update_path=True, download=True,
              'datasets/foo.tgz',
         misc='https://codeload.github.com/mne-tools/mne-misc-data/'
              'tar.gz/%s' % releases['misc'],
-        sample="https://osf.io/86qa2/download",
+        sample="https://osf.io/j4ms3/download",
```
Comment (larsoner) [inline mne/datasets/utils.py]: Instead of doing this, we should add the `mne.datasets.fetch_fsaverage_3layer_bem` function. We want it anyway so people don't need to download 1.5GB+ of data for this functionality, and the recent dataset update is problematic.

**Comment by massich** [comment]
This PR now goes after #6163

File: tutorials/plot_eeg_no_mri.py [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+"""  # noqa: E501
+
+# Authors: Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#          Joan Massich <mailsik@gmail.com>
+#
+# License: BSD Style.
+
+import os.path as op
+import mne
+from mne.datasets.sample import data_path
```
Comment (agramfort) [inline tutorials/plot_eeg_no_mri.py]: We should use sample here

Comment (larsoner) [reply tutorials/plot_eeg_no_mri.py]: I'll change it

**Comment by larsoner** [comment]
Changed to use `sample` data, made the `fsaverage` fetcher/updater, added to docs.

The localization isn't great, but we warned that it wouldn't be. Feel free to tweak until you're happy @agramfort @massich

File: tutorials/plot_eeg_no_mri.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+# License: BSD Style.
+
+import os.path as op
+
+import mne
+from mne.datasets import set_montage_coreg_path, fetch_fsaverage, sample
+
+# Convenience function to set ``subjects_dir`` default value for users who
+# only ever plan to do montage-based coreg with fsaverage.
+subjects_dir = set_montage_coreg_path()
```
Comment (larsoner) [inline tutorials/plot_eeg_no_mri.py]: For users with `SUBJECTS_DIR` set, this does nothing. For EEG users without such a value, `SUBJECTS_DIR` in MNE config gets set to `~/mne_data/MNE-fsaverage-data`.

File: tutorials/plot_eeg_no_mri.py [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+# Download fsaverage files
+fs_dir = fetch_fsaverage(verbose=True)
```
Comment (larsoner) [inline tutorials/plot_eeg_no_mri.py]: For users with `fsaverage` already, they will get theirs updated to have the latest files. On CircleCI, which sets `SUBJECTS_DIR` to the sample dataset directory, this looks like:
```
438 files missing from fsaverage in /home/circleci/mne_data/MNE-sample-data/subjects/fsaverage
Downloading missing files remotely
Downloading https://files.osf.io/v1/resources/rxvq7/providers/osfstorage/5cb759eea3bc970017f17d50?action=download&version=1&direct&revision=1 (536.0 MB)

Verifying hash 614a3680dcfcebd5653b892cc1234a4a.
File saved as /tmp/tmpihd8a9hb/fsaverage.zip.

Extracting missing files
Successfully extracted 438 files
```

**Comment by larsoner** [comment]
https://12307-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/plot_eeg_no_mri.html

**Comment by larsoner** [comment]
This pull request **introduces 1 alert** when merging f8224fe94cf0071873bed30dda6e1617f86c2549 into 2c215d8f61e79f86a908da7f9e995819c5d62ddf - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-a1bd467485a4ba9374ce42e2841451d27f57ea35)

**new alerts:**

* 1 for Use of the return value of a procedure

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by massich** [comment]
https://12312-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/plot_eeg_no_mri.html?highlight=plot_eeg_no_mri

**Comment by larsoner** [comment]
This pull request **introduces 1 alert** when merging 9f7cdf3d6bb22b02d41c01f9d821150ee054857f into aa7b3a0ba1038e6ad7fb357e60844696e234c40b - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-f2953eeb46689e39649a71a0a43edc5d46908df8)

**new alerts:**

* 1 for Use of the return value of a procedure

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: tutorials/plot_eeg_no_mri.py (hunk 12 above)
Comment (agramfort) [reply tutorials/plot_eeg_no_mri.py]: @larsoner to make both you and us (with @massich) happy I would suggest to create a function

mne.datasets.fsaverage.data_path()

that uses set_montage_coreg_path and fetch_fsaverage internally. The reason is that EEG people using a template should not bother about the subjects_dir and freesurfer jargon.

also you switched the example to use sample which uses fif files. I had proposed to use eegbci because it's edf files which are standard for EEG people. It feels more natural.

Any objection to these 2 points @larsoner ?

Comment (larsoner) [reply tutorials/plot_eeg_no_mri.py]: > mne.datasets.fsaverage.data_path()

Maybe just `mne.datasets.fetch_fsaverage(subjects_dir=None, update_path=True)`, and `update_path=True` will set the path using the `set_montage...` function. And we can even then make the `set_montage...` thing private.

I prefer the `fetch_` nomenclature here instead of `data_path` since it does something closer to what the existing `fetch_` functions do than to what the existing `data_path` functions do.

> I had proposed to use eegbci because it's edf files which are standard for EEG people. It feels more natural.

Above you said ["We should use sample here"](https://github.com/mne-tools/mne-python/pull/6146#discussion_r276330433) which is why I switched to the `sample` dataset. EEGBCI is fine with me, though showing a field map is a particularly uninteresting example, and something users are unlikely to want to do. It would be much better to, say, compute some classifier and then project it to source space or something. I can do it at the sprint next week.

File: tutorials/plot_eeg_no_mri.py (hunk 10 above)
Comment (agramfort) [reply tutorials/plot_eeg_no_mri.py]: oups my bad I typed to quickly and forgot the **not** ... oups...

**Comment by agramfort** [comment]
>
> Maybe just mne.datasets.fetch_fsaverage(subjects_dir=None,
> update_path=True), and update_path=True will set the path using the
> set_montage... function. And we can even then make the set_montage...
> thing private.
>

I like that. Make it private and make sure fetch_fsaverage works with a
good default path without thinking about it.


> I prefer the fetch_ nomenclature here instead of data_path since it does
> something closer to what the existing fetch_ functions do than to what
> the existing data_path functions do.
>
> I had proposed to use eegbci because it's edf files which are standard for
> EEG people. It feels more natural.
>
> Above you said "We should use sample here"
> <https://github.com/mne-tools/mne-python/pull/6146#discussion_r276330433>
> which is why I switched to the sample dataset. EEGBCI is fine with me,
> though showing a field map is a particularly uninteresting example, and
> something users are unlikely to want to do. It would be much better to,
> say, compute some classifier and then project it to source space or
> something. I can do it at the sprint next week.
>
my bad. I meant "not use sample" ie just use an fsaverage template dataset.
I suggested to do sensitivity maps to display something. Then the example
should not document the full pipeline but point to the narrative on making
epochs and inverse operators. This is to avoid a scattered doc that is
harder to maintain.

**Comment by larsoner** [comment]
I'm okay with the sensitivity map, we should just link at the end to a standard MNE tutorial or something then

**Comment by agramfort** [comment]
yes +100

**Comment by massich** [comment]
does it mean that we make #6167 green (or cherry pick 32efc818e517b6b0ab2dee89c18664a9f6741392) and call it a day? Or are we missing something else?

**Comment by massich** [comment]
> does it mean that we make #6167 green (or cherry pick 32efc81) and call it a day? Or are we missing something else?

I'm referring to the example itself.

**Comment by agramfort** [comment]
Yes. Lets move on

**Comment by larsoner** [comment]
Feel free to push to my branch the commit to go back to eegbci

File: mne/datasets/_fsaverage.py [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+    from .utils import _get_path
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=False)
+    if subjects_dir is None:
+        subjects_dir = _get_path(None, '', 'montage coregistration')
+        subjects_dir = op.join(subjects_dir, 'MNE-fsaverage-data')
+        os.makedirs(subjects_dir, exist_ok=True)
+    return subjects_dir
+
+
+def set_montage_coreg_path(subjects_dir=None):
```
Comment (agramfort) [inline mne/datasets/_fsaverage.py]: I would make this private

File: mne/bem.py [hunk 14]
```diff
@@ -253,16 +253,19 @@ def _fwd_bem_ip_modify_solution(solution, ip_solution, ip_mult, n_tri):
...
-                              ', '.join(str(f) for f in fewer)))
+        msg = ('Surface %s has topological defects: %d / %d '
+               'vertices have fewer than three neighboring '
+               'triangles [%s]' % (_surf_name[surf['id']],
+                                   len(fewer), surf['ntri'],
+                                   ', '.join(str(f) for f in fewer)))
+        if incomplete == 'raise':
+            raise RuntimeError(msg)
+        else:
+            warn(msg)
```
Comment (agramfort) [inline mne/bem.py]: why is this here???

File: mne/commands/tests/test_commands.py [hunk 15]
```diff
@@ -85,24 +85,24 @@ def test_clean_eog_ecg():
...
+        fid.write('MEG 2443\n')
+    shutil.copyfile(raw_fname, use_fname)
+    with ArgvSetter(('-i', use_fname, '--bad=' + bad_fname,
+                     '--rej-eeg', '150')):
+        with pytest.warns(None):  # samples, sometimes
+            fun.run()
+    fnames = glob.glob(op.join(tempdir, '*proj.fif'))
+    assert len(fnames) == 1
+    fnames = glob.glob(op.join(tempdir, '*-eve.fif'))
+    assert len(fnames) == 1
```
Comment (agramfort) [inline mne/commands/tests/test_commands.py]: how did this happen in this PR?

File: mne/bem.py (hunk 14 above)
Comment (mmagnuski) [reply mne/bem.py]: got here likely by accident in 8d1023ee13da529940e50949eb06e9ee0e94594b

File: mne/commands/mne_watershed_bem.py [hunk 16]
```diff
@@ -37,9 +37,6 @@ def run():
                       action="store_true")
     parser.add_option("-p", "--preflood", dest="preflood",
                       help="Change the preflood height", default=None)
-    parser.add_option("--copy", dest="copy",
-                      help="Use copies instead of symlinks for surfaces",
-                      action="store_true")
```
Comment (massich) [inline mne/commands/mne_watershed_bem.py]: @larsoner all this bem related things that were added to the PR. Were they needed?? I mean, they are unrelated to the PR. But maybe you wanted them.

Comment (larsoner) [reply mne/commands/mne_watershed_bem.py]: They were needed when I created the 3-layer BEM for fsaverage, so yes please keep

File: mne/commands/tests/test_commands.py [hunk 17]
```diff
@@ -233,13 +238,13 @@ def test_watershed_bem(tmpdir):
...
 @sample.requires_sample_data
-def test_flash_bem(tmpdir):
+def test_flash_bem():
     """Test mne flash_bem."""
     check_usage(mne_flash_bem, force_help=True)
     # Using the sample dataset
     subjects_dir = op.join(sample.data_path(download=False), 'subjects')
     # Copy necessary files to tempdir
-    tempdir = str(tmpdir)
+    tempdir = _TempDir()
```
Comment (larsoner) [inline mne/commands/tests/test_commands.py]: We want almost everything from this commit. Feel free to push a new PR but we shouldn't lose them

Comment (massich) [reply mne/commands/tests/test_commands.py]: no prob. I'll get it back and just ping you to make sure I did not forget anything important.

Comment (massich) [reply mne/commands/tests/test_commands.py]: Actually I made a PR https://github.com/mne-tools/mne-python/pull/6192 feel free to push in it.

**Comment by jona-sassenhagen** [comment]
Didn't see this ... very cool MNE is making an effort to support the "lazy and stupid" crowd finally!!!

File: tutorials/plot_eeg_no_mri.py [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+from mne.datasets import fetch_fsaverage
+
+# Download fsaverage files
+fs_dir = fetch_fsaverage(verbose=True)
+subjects_dir = op.dirname(fs_dir)
+
+# The files live in:
+subject = 'fsaverage'
+trans = op.join(fs_dir, 'bem', 'fsaverage-trans.fif')
+src = op.join(fs_dir, 'bem', 'fsaverage-5-src.fif')
```
Comment (drammock) [inline tutorials/plot_eeg_no_mri.py]: I think this needs to be `'fsaverage-ico-5-src.fif`?

**Comment by massich** [comment]
the failing error is unrelated. Merging
